js/script.js

```diff
@@ -19,22 +25,40 @@ const addButtonHandler = () => {
   }
 };
 
+const searchButtonHandler = () => {
+  const searchTitle = sanitizeInput($searchInput.value);
+
+  if (searchTitle) {
+    const filteredTasks = tasks.filter((task) =>
+      task.title.toLowerCase().includes(searchTitle.toLowerCase())
+    );
+    if (filteredTasks.length > 0) {
+      renderTasks(filteredTasks);
+    } else {
+      showToastMessage("No tasks found matching the search.");
+    }
+  } else {
+    renderTasks(tasks);
+  }
+  $searchInput.value = "";
+};
+
```

Comment:
we can make this more readable and clean

```suggestion
const handleSearchTasks = () => {
  // Sanitize and retrieve the input value
  const searchTitle = sanitizeInput($searchInput.value.trim());

  // Clear the input field
  $searchInput.value = "";

  // Use early return to handle empty search scenario
  if (!searchTitle) {
    renderTasks(tasks);
    return;
  }

  // Filter tasks based on the sanitized input
  const filteredTasks = tasks.filter(task => 
    task.title.toLowerCase().includes(searchTitle.toLowerCase())
  );

  // Render based on the filtering result
  if (filteredTasks.length > 0) {
    renderTasks(filteredTasks);
  } else {
    showToastMessage("No tasks found matching the search.");
  }
};


```

Reply:
Done, bhai.